Firestore/core/src/firebase/firestore/core/firestore_client.h

```diff
@@ -88,7 +88,8 @@ class FirestoreClient : public std::enable_shared_from_this<FirestoreClient> {
    * Terminates this client, cancels all writes / listeners, and releases all
    * resources.
    */
-  void Terminate(util::StatusCallback callback);
+  void TerminateAsync(util::StatusCallback callback);
+  void Terminate();
```

Comment:
Can you add a comment explaining how `Terminate` is different?

Reply:
It's a synchronous variant; I've added a comment.

Reply:
Thanks. Consider renaming the new function to `TerminateBlocking`; I think it would really highlight the difference.

Reply:
I'm partial to the convention of FooAsync/Foo pairs. This is already common in C# and C++ APIs to some extent.